tests/integration/renderer/test_dependencies.py

```diff
@@ -0,0 +1,47 @@
+from multiprocessing import Value
+
+import dash_core_components as dcc
+import dash_html_components as html
+import dash
+from dash.dependencies import Input, Output
+
+
+def test_rddp001_dependencies_on_components_that_dont_exist(dash_duo):
+    app = dash.Dash(__name__)
+    app.layout = html.Div(
+        [dcc.Input(id="input", value="initial value"), html.Div(id="output-1")]
+    )
+
+    output_1_call_count = Value("i", 0)
+
+    @app.callback(Output("output-1", "children"), [Input("input", "value")])
+    def update_output(value):
+        output_1_call_count.value += 1
+        return value
+
+    # callback for component that doesn't yet exist in the dom
+    # in practice, it might get added by some other callback
+    app.config.supress_callback_exceptions = True
+    output_2_call_count = Value("i", 0)
+
+    @app.callback(Output("output-2", "children"), [Input("input", "value")])
+    def update_output_2(value):
+        output_2_call_count.value += 1
+        return value
+
+    dash_duo.start_app_server(app)
+
+    assert dash_duo.find_element("#output-1").text == "initial value"
+    assert output_1_call_count.value == 1 and output_2_call_count.value == 0
+    dash_duo.percy_snapshot(name="dependencies")
+
+    dash_duo.find_element("#input").send_keys("a")
+    assert dash_duo.find_element("#output-1").text == "initial valuea"
+
+    assert output_1_call_count.value == 2 and output_2_call_count.value == 0
+
+    rqs = dash_duo.redux_state_rqs
+    assert len(rqs) == 1
+    assert rqs[0]["controllerId"] == "output-1.children" and not rqs[0]['rejected']
```

Comment:
I don't like the abstraction we did before `assertion_request_queue`,  it encapsulates  the checkpoints elsewhere, and someone also introduces a bug by inserting another parameter, but forget to check all the API calls, so we end up having several API calls to check the length of the queue, but it actually passed to the first added parameter which is a boolean. 

perhaps the only acceptable exception to hide the assertion with an API is the check below about clear browser logs.